lib/hysync/marc_synchronizer/runner.rb

```diff
@@ -137,22 +137,14 @@ def create_or_update_hyacinth_record(marc_record, base_digital_object_data, forc
             Rails.logger.error msg
           end
         elsif results.length == 1
-          pid = results.first['pid']
+          hyc_record = results.first
           # We want to preserve any existing identifiers from the existing item.
-          marc_hyacinth_record.digital_object_data['identifiers'].push(*(results.first['identifiers']))
-          marc_hyacinth_record.digital_object_data['identifiers'].uniq!
+          reconcile_identifiers!(marc_hyacinth_record, hyc_record)
```

Comment:
I pushed a commit to fix this line and the one below it.  I'm assuming you won't object, but just wanted to highlight the change here.

Reply:
Hmm. This suggests a significant hole in the test coverage.